core/shared/src/main/scala/com/monovore/decline/opts.scala

```diff
@@ -133,6 +134,33 @@ object Opts {
     Single(Opt.Regular(namesFor(long, short), metavarFor[A](metavar), help, visibility))
       .mapValidated(Argument[A].read)
 
+  def optionalOptArg[A](
```

Comment:
Definitely a nitpick, but I'm not sure about the name. (Mix of full words and abbreviations, some of which aren't used elsewhere.) I'm not sure what else I'd suggest, though: maybe `optionalOption`, or `optionFlag` or `flagOption` to suggest you're getting both option-like and flag-like behaviours?